tools/layer-zero-example/contracts/HTSConnector.sol

```diff
@@ -0,0 +1,128 @@
+     * @param _name The name of the OFT.
+     * @param _symbol The symbol of the OFT.
+     * @param _lzEndpoint The LayerZero endpoint address.
+     * @param _delegate The delegate capable of making OApp configurations inside of the endpoint.
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        address _lzEndpoint,
+        address _delegate
+    ) payable OFTCore(8, _lzEndpoint, _delegate) {
+        IHederaTokenService.TokenKey[] memory keys = new IHederaTokenService.TokenKey[](2);
+        keys[0] = getSingleKey(
+            KeyType.ADMIN,
+            KeyType.PAUSE,
+            KeyValueType.INHERIT_ACCOUNT_KEY,
+            bytes("")
+        );
+        keys[1] = getSingleKey(
+            KeyType.SUPPLY,
+            KeyValueType.INHERIT_ACCOUNT_KEY,
+            bytes("")
+        );
+
+        IHederaTokenService.Expiry memory expiry = IHederaTokenService.Expiry(0, address(this), 8000000);
+        IHederaTokenService.HederaToken memory token = IHederaTokenService.HederaToken(
+            _name, _symbol, address(this), "memo", true, 5000, false, keys, expiry
+        );
+
+        (int responseCode, address tokenAddress) = HederaTokenService.createFungibleToken(
+            token, 1000, int32(int256(uint256(8)))
+        );
+        require(responseCode == HederaResponseCodes.SUCCESS, "Failed to create HTS token");
+
+        int256 transferResponse = HederaTokenService.transferToken(tokenAddress, address(this), msg.sender, 1000);
+        require(transferResponse == HederaResponseCodes.SUCCESS, "HTS: Transfer failed");
+
+        htsTokenAddress = tokenAddress;
+
+        emit CreatedToken(tokenAddress);
+    }
+
+    /**
+     * @dev Retrieves the address of the underlying ERC20 implementation.
+     * @return The address of the OFT token.
+     *
+     * @dev In the case of OFT, address(this) and erc20 are the same contract.
+     */
+    function token() public view returns (address) {
+        return address(this);
+    }
+
+    /**
+     * @notice Indicates whether the OFT contract requires approval of the 'token()' to send.
+     * @return requiresApproval Needs approval of the underlying token implementation.
+     *
+     * @dev In the case of OFT where the contract IS the token, approval is NOT required.
+     */
+    function approvalRequired() external pure virtual returns (bool) {
+        return false;
+    }
+
+
+    /**
+     * @dev Burns tokens from the sender's specified balance.
+     * @param _from The address to debit the tokens from.
+     * @param _amountLD The amount of tokens to send in local decimals.
+     * @param _minAmountLD The minimum amount to send in local decimals.
+     * @param _dstEid The destination chain ID.
+     * @return amountSentLD The amount sent in local decimals.
+     * @return amountReceivedLD The amount received in local decimals on the remote.
+     */
+    function _debit(
+        address _from,
```

Comment:
Q: If `_from` is not msg.sender then it presumes an approval exists for the contract right?
If not do we need a check?

Reply:
There was a check here and approval that had been executed afterward but after some feedback from Fernando, I decided to completely remove the approval from this call. It's user's/dapp's responsibility to ensure the token's approval beforehand like it's done in ERC20 swaps via 2 transactions so we don't need any extra actions here.